app/forms/candidate_interface/degree_wizard.rb

```diff
@@ -98,50 +98,9 @@ def other_grade
     end
 
     def next_step(step = current_step)
-      if !reviewing? || country_changed?
-        if step == :country && uk?
-          :degree_level
-        elsif (step == :type && international?) || step == :degree_level
-          :subject
-        elsif (step == :subject && uk? && !degree_has_type?) || (step == :type && uk?) || (step == :subject && international?)
-          :university
-        elsif (step == :country && international? && country.present?) || (step == :subject && uk?)
-          :type
-        elsif step == :university
-          :completed
-        elsif (step == :completed && phd?) || step == :grade
-          :start_year
-        elsif step == :completed
-          :grade
-        elsif step == :start_year
-          :award_year
-        elsif step == :award_year && international? && completed?
-          :enic
-        elsif step == :enic && international? && enic_reason == HAS_STATEMENT
-          :enic_reference
-        elsif %i[award_year enic enic_reference].include?(step)
-          :review
-        else
-          Sentry.capture_exception(
-            InvalidStepError.new(
-              "Invalid Step for application_form: #{application_form_id}, previous_step: #{previous_step}",
-            ),
-          )
-          rescue_step
-        end
-      elsif step == :degree_level && degree_has_type?
-        :type
-      elsif step == :completed
-        :award_year
-      elsif step == :award_year && completed? && international?
-        :enic
-      elsif step == :enic && international? && enic_reason != HAS_STATEMENT
-        :review
-      elsif step == :enic && international?
-        :enic_reference
-      else # rubocop:disable Lint/DuplicateBranch
-        :review
-      end
+      return uk? ? uk_steps(step) : international_steps(step) if !reviewing? || country_changed?
```

Comment:
mixing a ternary with an early return, which has two conditions is cognitively confusing / difficult to read. Perhaps instead:
```
if !reviewing? || country_changed?
  uk? ? uk_steps(step) : international_steps(step)
else
  reviewing_or_country_not_changed_steps(step)
end
```

Reply:
I agree that this can be cognitively confusing. I have changed it to your version. Thanks 👍🏼 

https://github.com/DFE-Digital/apply-for-teacher-training/pull/10142/commits/1bc96d6dfc5f508215cda766712e20aab09ccc69

Reply:
Did rubocop change it back?

Reply:
No, I don't think so?

Isn't this your suggestion?

https://github.com/DFE-Digital/apply-for-teacher-training/pull/10142/files#diff-50ba5523c917aa72fd8bcc6004fea7b7bb8691b6bc9b0d9d1ee29a7adbbac566R100-R104